keel-api/src/test/resources/examples/alb-example.yml

```diff
@@ -4,14 +4,13 @@ kind: application-load-balancer
 metadata:
   serviceAccount: my-email@spinnaker.io
 spec:
-  vpcName: vpc0
-  subnetType: internal (vpc0)
   moniker:
     app: keeldemo
     stack: example
     detail: albec2v1
   locations:
     accountName: test
+    vpcName: vpc0
     regions:
     - region: us-east-1
       subnet: internal (vpc0)
```

Comment:
Generally moving `vpcName` from the root of the spec into the `locations` and `subnet` to each individual region.

Reply:
Do we foresee a case where a user would deploy the same thing in subnets with different purposes in different regions (say `internal (vpc0)` in `us-east-1` and `external (vpc0)` in `us-west-2`)? Seems unlikely...

Reply:
In fact, we already do that for load balancers in the UI with the guided setup:
![image](https://user-images.githubusercontent.com/1323478/66859385-0c035080-ef40-11e9-9b05-36d30b10b79d.png)

Reply:
You mean `subnet` could be moved up from the individual region to sit next to `vpcName`? What do you think @asher ?

Reply:
This seems right. The resolved subnet might differ across both regions and individual AZ's, but all should map back to the same named purpose.

Reply:
don't have an informed opinion on whether subnet should be regional or not in these specs, but worth thinking about whether abstracting up to `internal` vs. `external` like we do in Luis's UI shots should really be at this level or at another level that can automatically resolve all the appropriate subnets for each region. Those two threads of conversation might be separate.

Reply:
nit: See, @asher even said it himself above: what we're naming in the spec is basically the purpose of the subnet. 😃 Anyway, I don't have an issue with keeping the subnet names in the spec (as opposed to the more abstract "purpose"), but in that case I'd suggest at least renaming the field to `subnetName` for consistency, or removing the `Name` suffix from `vpcName`.

Reply:
P.S. @asher Just enlightened me that the value we expect in that `subnet` field is actually the value of the `purpose` metadata tag attached to the resource. So my last point above is moot. Wondering if it would be best to name the property `subnetPurpose` for clarity, though...

Reply:
If you hit the CloudDriver endpoint you'll see that subnets don't have a "name" as such. The closest thing is the "purpose". But yeah, I'd be down with dropping `Name` from `vpcName`.